### PR TITLE
AmqpForwardAttribute mailet: ignore errors when creating an exchange that already exists with a different configuration.

### PR DESCRIPTION
**Pull request by vttranlina**
It will help us avoid starting James with failures. 
Additionally, with each parameter configuration (e.g., durable, autoDelete, internal, passive), we do not need to have the same argument settings in the mailetcontainer.

### CODE REVIEW

File: mailet/amqp/src/main/java/org/apache/james/transport/mailets/AmqpForwardAttribute.java [hunk 1]
```diff
@@ -154,7 +156,13 @@ public void init() throws MailetException {
                 .map(type -> ExchangeSpecification.exchange(exchange).type(type))
                 .orElse(ExchangeSpecification.exchange(exchange));
 
-            sender.declareExchange(exchangeSpecification).block();
+            sender.declareExchange(exchangeSpecification)
+                .onErrorResume(error -> error instanceof ShutdownSignalException && error.getMessage().contains("reply-code=406, reply-text=PRECONDITION_FAILED"),
+                    error -> {
+                        LOGGER.warn("Exchange `{}` already exists but with different configuration. Ignoring this error. \nError message: {}", exchange, error.getMessage());
+                        return Mono.empty();
+                    })
```
Comment (chibenwa) [inline mailet/amqp/src/main/java/org/apache/james/transport/mailets/AmqpForwardAttribute.java:164]: Please have it tested:

 - create an exchange with different settings
 - Start the mailet (throwns the exception here)
 - And process a mail and ensure the mail gets well processed
 
 On principle looks great in practice last time I tried the entire session was F... up... So I want to see it running before approving it...

**Comment by quantranhong1999** [comment]
Read it.

**Review by Arsnael — APPROVED** [review]

**Comment by vttranlina** [comment]
I renamed `AmqpForwardAttributeWithFanoutExchangeTypeIntegrationTest` -> `AmqpForwardAttributeIntegrationTest`
Refactor the code and include a test case for this scenario.

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Review by quantranhong1999 — APPROVED** [review]

**Comment by vttranlina** [comment]
```
Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.3.0:test (default-test) on project scaling-pulsar-smtp:
```
it does not relate to this pr, 
can you rerun ci?

**Pull request merged**